File: api/models/employee.py
from __future__ import annotations
from typing import TYPE_CHECKING, Optional
from sqlmodel import Relationship, SQLModel, Field

if TYPE_CHECKING:
    from models.user import User

class EmployeeBase(SQLModel):
    name: str = Field(index=True, description="姓名")
    gender: int = Field(default=0, index=True, description="0为其他, 1为男, 2为女")
    department: Optional[str] = Field(default=None, index=True, description="部门名称")
    level: int = Field(default=1, index=True, description="范围0-5")
    status: int = Field(default=0, index=True, description="0为在职, 1为离职,2为停用")

class Employee(EmployeeBase, table=True):
    employee_id: str = Field(
        default=None,
        primary_key=True,
        index=True,
        unique=True,
        description="工号")

    __tablename__ = "employees"
    '''
    user: Optional["User"] = Relationship(
        back_populates="employee", # 指向 User.employee
        sa_relationship_kwargs={"lazy": "selectin"}
    )
    '''

File: web/app/model/Employee.js
export default class Employee {
  constructor(data) {
    this.name = data.name;
    this.gender = data.gender;
    this.department = data.department;
    this.level = data.level;
    this.status = data.status;
    this.employee_id = data.employee_id;
  }

  toJson() {
    return {
      name: this.name,
      gender: this.gender,
      department: this.department,
      level: this.level,
      status: this.status,
      employee_id: this.employee_id,
    };
  }
}